### PR TITLE
Go: Add models-as-data content for pointer content

### PR DESCRIPTION
**Pull request by owen-mc**
This is based on https://github.com/github/codeql/pull/13453 and fixes the failing tests added in that PR.

### CODE REVIEW

**Review by smowton — DISMISSED** [review]

**Comment by michaelnebel** [comment]
- Even though it is currently not used for Go, I it is probably worth also updating the pretty printing predicates to support *Dereference*: https://github.com/github/codeql/blob/main/go/ql/lib/semmle/go/dataflow/internal/FlowSummaryImplSpecific.qll#LL91C16-L91C34 (as a part of this PR)
- Maybe it is also worth introducing a unit test (similar to the one that exists for C#) that prints summaries (this will use the query predicate: https://github.com/github/codeql/blob/main/go/ql/lib/semmle/go/dataflow/internal/FlowSummaryImpl.qll#L1377). (as a follow up PR as it requires a bit more work).

**Review by michaelnebel — COMMENTED** [review]
Really cool with an extension to the MaD DSL for GO :-)

File: go/ql/lib/semmle/go/dataflow/ExternalFlow.qll [hunk 1]
```diff
@@ -342,6 +342,8 @@ predicate parseContent(string component, DataFlow::Content content) {
   component = "MapKey" and content instanceof DataFlow::MapKeyContent
   or
   component = "MapValue" and content instanceof DataFlow::MapValueContent
+  or
+  component = "Dereference" and content instanceof DataFlow::PointerContent
```
[the remark this answers is not in the document]
Comment (owen-mc) [reply go/ql/lib/semmle/go/dataflow/ExternalFlow.qll:346]: Access paths weren't mentioned. I've copied what C# says about it and updated it to newer syntax without "of". I've taken the liberty of updating the C# one as well.

**Review by michaelnebel — APPROVED** [review]
Looks good to me and thank you for the C# update as well 👍

**Review by smowton — APPROVED** [review]

**Pull request merged**